server/modelpath.go

```diff
@@ -103,13 +103,16 @@ func (mp ModelPath) GetShortTagname() string {
 // modelsDir returns the value of the OLLAMA_MODELS environment variable or the user's home directory if OLLAMA_MODELS is not set.
 // The models directory is where Ollama stores its model files and manifests.
 func modelsDir() (string, error) {
-	if models, exists := os.LookupEnv("OLLAMA_MODELS"); exists {
-		return models, nil
-	}
 	home, err := os.UserHomeDir()
 	if err != nil {
 		return "", err
 	}
+	if models, exists := os.LookupEnv("OLLAMA_MODELS"); exists {
+		if strings.HasPrefix(models, "~/") {
```

Comment:
Should we allow windows paths here too?
```
$env:OLLAMA_MODELS="~\ollama-dir"
ollama.exe serve
```

Reply:
As far as I know Windows uses `%USERPROFILE%` rather than `~`, but good call, generally better to use os.PathSeparator

Reply:
In powershell "~" does work.